hathor/transaction/block.py

```diff
@@ -123,7 +123,7 @@ def create_from_struct(cls, struct_bytes: bytes,
 
     def calculate_height(self) -> int:
         """Return the height of the block, i.e., the number of blocks since genesis"""
-        if self.is_genesis:
+        if not self.parents:
```

Comment:
The `is_genesis` is a method, so I had to use another verification to skip the genesis. This is one of the dangers of using `@property` decorator.

Reply:
Being a property shouldn't matter much, but from a comment below I get that it's because it uses the storage, but I think we could do away with depending on the storage, since using the storage is being more trouble than being useful.

Reply:
I didn't understand your suggestion. I agree that it would be better to use `if self.is_genesis:`, and I'll check what part breaks when I do it.

Reply:
What I suggested was:

```python
if not self.parents:
    assert self.is_genesis
    ...
```

But now I understand that the change of the `if` condition from `self.is_genesis` to `not self.parents` is precisely because `self.is_genesis` needs a storage. If that's the case, fixing `self.is_genesis` seems like it would make it possible to use that for the `if` condition. Which makes a little more sense to me, but it's probably subjective anyway. Looking back it doesn't seem that important.

Reply:
I agree. I just reverted the change. Would you check again, please?

Reply:
Great!